### PR TITLE
Fix/payp 890

### PR DESCRIPTION
**Pull request by askur87**
- Fixing NPE when CustomerDetail object not set and CustomerIdentifier parameter not set.
- Support for Android 11 to get telephonyManager.getDataNetworkType()

### CODE REVIEW

File: demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java [hunk 1]
```diff
@@ -398,7 +398,10 @@ private void initData() {
     }
 
     private CustomerDetails getUserDetail() {
-        return MidtransSDK.getInstance().getTransactionRequest().getCustomerDetails();
+        if (MidtransSDK.getInstance().getTransactionRequest().getCustomerDetails() != null) {
+            return MidtransSDK.getInstance().getTransactionRequest().getCustomerDetails();
+        }
+        return null;
```
Comment (KarthikUCH) [inline demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java:404]: I think its better to undo the change, since the previous logic also return null

File: uikit/src/main/java/com/midtrans/sdk/uikit/activities/PaymentMethodsActivity.java [hunk 2]
```diff
@@ -307,14 +307,14 @@ private void getPaymentPages() {
...
             }
 
-            CustomerDetails userDetail = midtransSDK.getTransactionRequest().getCustomerDetails();
-            if (userDetail == null) {
-                userId = UUID.randomUUID().toString();
-            } else {
-                userId = (userDetail.getCustomerIdentifier().isEmpty()) ? UUID.randomUUID().toString() : userDetail.getCustomerIdentifier();
+            if (midtransSDK.getTransactionRequest().getCustomerDetails() != null) {
+                CustomerDetails customerDetails = midtransSDK.getTransactionRequest().getCustomerDetails();
+                if (customerDetails != null && customerDetails.getCustomerIdentifier() != null) {
```
Comment (KarthikUCH) [inline uikit/src/main/java/com/midtrans/sdk/uikit/activities/PaymentMethodsActivity.java:312]: Suggest removing `customerDetails != null` check, since 
`if (midtransSDK.getTransactionRequest().getCustomerDetails() != null) ` already made. 

If not remove previous if condition `if (midtransSDK.getTransactionRequest().getCustomerDetails() != null) {`

File: demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java (hunk 1 above)
Comment (askur87) [reply demo/src/main/java/com/midtrans/demo/DemoOrderReviewActivity.java:404]: Yes you are correct, will revert it

File: uikit/src/main/java/com/midtrans/sdk/uikit/activities/PaymentMethodsActivity.java (hunk 2 above)
Comment (askur87) [reply uikit/src/main/java/com/midtrans/sdk/uikit/activities/PaymentMethodsActivity.java:312]: Yes you are correct, will remove it

**Comment by KarthikUCH** [comment]
LGTM

**Review by theanh0512 — APPROVED** [review]

**Comment by theanh0512** [comment]
LGTM

**Review by uziwuzzy — APPROVED** [review]
